src/unit_tests/os_crypto/sha512/test_sha512_op.c

```diff
@@ -45,11 +44,37 @@ void test_sha512_hex() {
     assert_string_equal(buffer, string_sha512);
 }
 
+void test_sha512_file() {
+    const char *string = "teststring";
+    const char *string_sha512 = "6253b39071e5df8b5098f59202d414c37a17d6a38a875ef5f8c7d89b0212b028692d3d2090ce03ae1de66c862fa8a561e57ed9eb7935ce627344f742c0931d72";
+    os_sha512 buffer;
+
+    /* create tmp file */
+    char file_name[256];
+    strncpy(file_name, "/tmp/tmp_file-XXXXXX", 256);
+    int fd = mkstemp(file_name);
+
+    write(fd, string, strlen(string));
+    close(fd);
+
+    OS_SHA512_File(file_name, buffer, OS_TEXT);
```

Comment:
Returned value of the function OS_SHA512_File could be tested to be 0 (zero)

Reply:
done